src/mcedit2/panels/worldmeta.py

```diff
@@ -0,0 +1,255 @@
+        with command.begin():
+            self.worldMeta.generatorName = self.GENERATOR_TYPES[self.generationTypeCombo.currentIndex()]
+        self.editorSession.pushCommand(command)
+
+    def spawnXChanged(self):
+        if self.editsDisabled:  # xxx copied from inventory.py
+            return
+
+        command = WorldMetaEditCommand(self.editorSession, self.tr("Change Spawn X"))
+        with command.begin():
+            self.worldMeta.SpawnX = self.spawnX.value()
+        self.editorSession.pushCommand(command)
+
+    def spawnYChanged(self):
+        if self.editsDisabled:  # xxx copied from inventory.py
+            return
+
+        command = WorldMetaEditCommand(self.editorSession, self.tr("Change Spawn Y"))
+        with command.begin():
+            self.worldMeta.SpawnY = self.spawnY.value()
+        self.editorSession.pushCommand(command)
+
+    def spawnZChanged(self):
+        if self.editsDisabled:  # xxx copied from inventory.py
+            return
+
+        command = WorldMetaEditCommand(self.editorSession, self.tr("Change Spawn Z"))
+        with command.begin():
+            self.worldMeta.SpawnZ = self.spawnZ.value()
+        self.editorSession.pushCommand(command)
+
+    def timeChanged(self):
+        if self.editsDisabled:  # xxx copied from inventory.py
+            return
+
+        command = WorldMetaEditCommand(self.editorSession, self.tr("Change DayTime"))
+        with command.begin():
+            days, hours, minutes = self.timeDays.value(), self.timeHours.value(), self.timeMinutes.value()
+            time = max((days * 24000 + hours * 1000 + int(minutes * (1000.0 / 60.0))) - 30000, 0)
+            self.worldMeta.DayTime = time
+        self.editorSession.pushCommand(command)
+
+    def hardcoreChanged(self):
+        if self.editsDisabled:  # xxx copied from inventory.py
+            return
+
+        command = WorldMetaEditCommand(self.editorSession, self.tr("Change Hardcore"))
+        with command.begin():
+            self.worldMeta.hardcore = self.hardcoreBool.isChecked()
+        self.editorSession.pushCommand(command)
+
+    def lockedDifficultyChanged(self):
+        if self.editsDisabled:  # xxx copied from inventory.py
+            return
+
+        command = WorldMetaEditCommand(self.editorSession, self.tr("Change Locked Difficulty"))
+        with command.begin():
+            self.worldMeta.DifficultyLocked = self.lockedDifficultyBool.isChecked()
+        self.editorSession.pushCommand(command)
+
+    def allowCommandsChanged(self):
+        if self.editsDisabled:  # xxx copied from inventory.py
+            return
+
+        command = WorldMetaEditCommand(self.editorSession, self.tr("Change Allow Commands"))
+        with command.begin():
+            self.worldMeta.allowCommands = self.commandsBool.isChecked()
+        self.editorSession.pushCommand(command)
+
+class WorldMetaEditCommand(NBTDataChangeCommand):
```

Comment:
NBTDataChangeCommand doesn't do anything special and in any case should only be used by the NBT editor widget. I'd inherit from SimpleRevisionCommand instead.